---
title: "METAGENOMICS PROJECT"
author: "Daniel Martín, Guillem Miró, Marc Pinós"
date: "2023-11-13"
output: pdf_document
editor_options: 
  markdown: 
    wrap: 72
---

# INTRODUCTION

Sequencing and metagenomics are biological disciplines that study the
genetic and epigenetic sequence information of organisms. By sequencing
their genome they attempt to understand several genetic concepts such
as, genetic functionality, genotype, phenotype, and inheritance among
others.

In this project a metagenomic analysis will be performed using an
initial sequenced data set of an unknown bacterial species as reference.
By performing an exhaustive methodology, latter sequence will be sorted
out in order to identify its origin and produce several analysis.

## Objectives

-   To perform a metagenomic analysis from a genomic sequence of unknown
    origin.
-   To select relevant and higher quality information from the sequence
    in order to properly work with it.
-   To identify the taxonomic classification of the unknown sequence.
-   To infer functional categories of the genomic protein sequences and
    analyze a selected category.
-   To produce a phylogeny of the reference genome and its nearer
    species.

# METHODS

## Practical 4: Assembly of a bacterial genome

Raw sequence data will be assembled and analyzed so as to purge for a
higher quality sample. In our particular case, sequencing data is
composed in a FASTQ format, which provides the sequence itself and its
quality information.

### 1. Read Quality Control
The sequence data will be analyzed by *Assess Read Quality with FastQC*
so as to make en exhaustive report of the reads' quality. This will
enable the detection of possible noise, unwanted/impure fragments,
sequencing mistakes, and base distribution rates.

*Parameters:*

-   Basic Statistics
-   Per base sequence quality
-   Per tile sequence quality
-   Per sequence quality scores
-   Per base sequence content
-   Per sequence GC content
-   Per base N content
-   Sequence length distribution
-   Sequence duplication levels

### 2. Trimming
Once all impure fragments are detected in the reads, trimming will be
performed. By using *Trim Reads with Trimmomatic* an input read will be
cut following a base pair (bp) threshold. For instance, starting and end
point can be stipulated by selecting crop reads, which selects the
number of bp to keep from start of the read.

The program will produce a new file which will correspond to the new
sequence already trimmed.

### 3. Read Quality Control
The trimmed sequence will be assessed a new quality control (using
*Assess Read Quality with FastQC* again) so as to identify all the
compensations of quality compared to the raw analysis.

If all the previously-identified mistakes are compensated by the
trimming, the analysis will be continued.

### 4. Assemble Read
All the reads will be stored into continuous sequences. In our case,
*MEGAHIT* will be used. Compared to other softwares it is believed to be
faster. All the reads will be conjugated to create long DNA fragments,
known as *contigs* (assemble continuous genome fragments).

### 5. Binning
Previous *contigs* will be grouped in several *bins*. The term bin
corresponds to a putative population genomes created by a selection of
properties, such as codon usage, genome related statistics, GC
proportion, read length, among other parameters. The analysis will be
performed using *MaxBin2*.

### 6. Bin quality control
Finally, all bins created will be processed under a quality control. It
will provide an estimate of genome completeness and contamination by
plotting each bin compared to its expected distribution of a typical
genome. *CheckM* will be used to perform the control.

*In particular, we will look for high quality genomes, which correspond
to \>90% completeness and \<5% of contamination.*

## Practical 5: Bacterial genome identification

Once all quality reads are selected, the analysis must begin. In
particular a taxonomic identification of out unknown sample will be
performed.

In the case of bacteria, as they do not reproduce sexually, species
classification is induced by the computation of ANI (average nucleotide
identity). Bacteria are known to exchange core genes between mainly same
species, and being unusual (or at least more difficult) to pass to
different ones. Therefore, computing percentage of identity between a
reference species and the sample will enable an identity classification:

gANI:

-   95% same species

-   85% same genus

-   70% same family

### 1. Genome annotation
The higher quality bin from practical 4 will be selected to perform the
analysis. *PROKKA* software will be used to annotate the sample genome
(placed in the bin) in order to identify several pieces or regions prone
to be compared.

### 2. Genome set
Selecting latter annotated genome, we will create a genome set, enabilg
a further research with the sample.

### 3. Microbe classification
The genome set will be taxonomically classified using *GTDB*. As an
output a taxonomical tree will promote the identification of the
bacterial sepecies present in our sample.

### 4. GTDB: genome taxonomy data base
[GTDB](https://gtdb.ecogenomic.org/) will be used to download the
*Genebank assembly accession* of out bacteria and acquire a FASTA
reference sequence.

### 5. ANI computation
Latter FASTA sequence (type strain genome) and the annotated genome
(Meta genome assembled genome) from step 1 will be compared using
*FastANI*, which will compute the ANI.

In particular, an accurate ANI will be estimated by the comparison of
nucleotide of orthologous gene pairs shared between the 2 sequences
given.

## Practical 6: Genome annotation I
In practical 6 we will proceed with the genome annotation of both sample and refernce genome. 

### 1. Reference genome annotation
Following last practical's procedure, the genome extracted from [GTDB](https://gtdb.ecogenomic.org/) will be also annotated.

### 2. Data exploration and functional categories
 By selecting the annotated sample, we will select the binocular option in order to explore all data gathered. 

Thanks to _PROKKA annotation pipeline_ we will be able to acknowledge provenance, linked samples, genome overview, taxonomy, publications, and assembly and annotation from our sample. The latter category will be selected to find the functional categories encoded in our genome, which corresponds to a set of proteins prone to be expressed by our bacterial specimen. 

The functional categories will be compared to the reference genome estracted from [GTDB](https://gtdb.ecogenomic.org/).

Afterwards, we will download a GFF (_gene feature format_) and a FASTA files to work on both genomes. A specific functional category will be selected and we will compute:

- The number of tRNAs of our sample.
- The number of rRNAs of our sample.
- The number of genes encoding proteins without known function.

Finally, using _RStudio_ we will create a barplot of the types of genes found in our MAG and our genome of reference (type strain). 


## Practical 7: Genome annotation II
Genome annotation will be carried on. Having previously selected a functional category of our genome both nucleotide and amino acid sequence will be studied.

### 1. Multiple sequence alignment

### 2. Pairwise sequence alignment

## Practical 8

# RESULTS AND DISCUSSION

## Practical 4

The second quality control were assessed and showed a high quality score
in all base pair position of all short reads, even in the head and tail
fragments of each read. The result is shown in Figure XXX. REMARCAR QUE
NO HACEMOS TRIMMING.

## Practical 5

## Practical 6

## Practical 7

## Practical 8

# CONCLUSIONS

# REFERENCES
