---
title: "METAGENOMICS ASSEMBLY PROJECT"
author: "Daniel Martín, Guillem Miró, Marc Pinós"
date: "2023-11-23"
geometry: "left=2.5cm,right=2.5cm,top=2cm,bottom=2cm"
output:
  pdf_document:
header-includes:
- \usepackage{caption}
- \captionsetup[figure]{labelformat=empty}
editor_options: 
  markdown: 
    wrap: 72
---

# INTRODUCTION

Sequencing and metagenomics are biological disciplines that study the
genetic and epigenetic sequence information of organisms. By sequencing
their genome they attempt to understand several genetic concepts such
as, genetic functionality, genotype, phenotype, and inheritance among
others.

In this project a metagenomic analysis will be performed using an
initial sequenced data set of an unknown bacterial species (originally from fecal samples) as reference.
By performing an exhaustive methodology, latter sequence will be sorted
out in order to identify its origin and produce several analysis.

## Objectives

-   To perform a metagenomic analysis from a genomic sequence of unknown
    origin.
-   To select relevant and higher quality information from the sequence
    in order to properly work with it.
-   To identify the taxonomic classification of the unknown sequence.
-   To infer functional categories of the genomic protein sequences and
    analyze a selected category.
-   To produce a phylogeny of the reference genome and its nearer
    species.

# METHODS

## [Practical 4: Assembly of a bacterial genome]{.underline}

Raw sequence data will be assembled and analyzed so as to purge for a
higher quality sample. In our particular case, sequencing data is
composed in a FASTQ format, which provides the sequence itself and its
quality information.

### 1. Read Quality Control

The sequence data will be analyzed by *Assess Read Quality with FastQC*
so as to make en exhaustive report of the reads' quality. This will
enable the detection of possible noise, unwanted/impure fragments,
sequencing mistakes, and base distribution rates.

*Parameters:* a) Basic Statistics, b) Per base sequence quality, c) Per
tile sequence quality, d) Per sequence quality scores, e) Per base
sequence content, f) Per sequence GC content, g) Per base N content, h)
Sequence length distribution, i) Sequence duplication levels.

### 2. Trimming

Once all impure fragments are detected in the reads, trimming will be
performed. By using *Trim Reads with Trimmomatic* an input read will be
cut following a base pair (bp) threshold. For instance, starting and end
point can be stipulated by selecting crop reads, which selects the
number of bp to keep from start of the read.

The program will produce a new file which will correspond to the new
sequence already trimmed.

### 3. Read Quality Control

The trimmed sequence will be assessed a new quality control (using
*Assess Read Quality with FastQC* again) so as to identify all the
compensations of quality compared to the raw analysis.

If all the previously-identified mistakes are compensated by the
trimming, the analysis will be continued.

### 4. Assemble Read and binning

All the reads will be stored into continuous sequences. In our case,
*MEGAHIT* will be used. Compared to other softwares it is believed to be
faster. All the reads will be conjugated to create long DNA fragments,
known as *contigs* (assemble continuous genome fragments).

Previous *contigs* will be grouped in several *bins*. The term bin
corresponds to a putative population genomes created by a selection of
properties, such as codon usage, genome related statistics, GC
proportion, read length, among other parameters. The analysis will be
performed using *MaxBin2*.

### 5. Bin quality control

Finally, all bins created will be processed under a quality control. It
will provide an estimate of genome completeness and contamination by
plotting each bin compared to its expected distribution of a typical
genome. *CheckM* will be used to perform the control.

*In particular, we will look for high quality genomes, which correspond
to \>90% completeness and \<5% of contamination.*

## [Practical 5: Bacterial genome identification]{.underline}

Once all quality reads are selected, the analysis must begin. In
particular, a taxonomic identification of out unknown sample will be
performed.

In the case of bacteria, as they do not reproduce sexually, species
classification is induced by the computation of ANI (average nucleotide
identity). Bacteria are known to exchange core genes between mainly same
species, and being unusual (or at least more difficult) to pass to
different ones.

Therefore, computing percentage of identity between a reference species
and the sample will enable an identity classification by gANI
percentage: a) 95% same species; b) 85% same genus; c) 70% same family.

### 1. Genome annotation

The higher quality bin from practical 4 will be selected to perform the
analysis. *PROKKA* software will be used to annotate the sample genome
(placed in the bin) in order to identify several pieces or regions prone
to be compared.

### 2. Genome set and microbe classification

Selecting latter annotated genome, we will create a genome set, enabling
a further research with the sample. It will be taxonomically classified
using [GTDB](https://gtdb.ecogenomic.org/). As an output a taxonomical
tree will promote the identification of the bacterial sepecies present
in our sample.

### 3. GTDB: genome taxonomy data base

[GTDB](https://gtdb.ecogenomic.org/) will be used to download the
*Genebank assembly accession* of out bacteria and acquire a FASTA
reference sequence.

### 4. ANI computation

Latter FASTA sequence (type strain genome) and the annotated genome
(Meta genome assembled genome) from step 1 will be compared using
*FastANI*, which will compute the ANI.

In particular, an accurate ANI will be estimated by the comparison of
nucleotide of orthologous gene pairs shared between the 2 sequences
given.

## [Practical 6: Genome annotation I]{.underline}

In practical 6 we will proceed with the genome annotation of both sample
and refernce genome.

### 1. Reference genome annotation

Following last practical's procedure, the genome extracted from
[GTDB](https://gtdb.ecogenomic.org/) will be also annotated.

### 2. Data exploration and functional categories

By selecting the annotated sample, we will select the binocular option
in order to explore all data gathered.

Thanks to *PROKKA annotation pipeline* we will be able to acknowledge
provenance, linked samples, genome overview, taxonomy, publications, and
assembly and annotation from our sample. The latter category will be
selected to find the functional categories encoded in our genome, which
corresponds to a set of proteins prone to be expressed by our bacterial
specimen.

The functional categories will be compared to the reference genome
estracted from [GTDB](https://gtdb.ecogenomic.org/).

Afterwards, we will download a GFF (*gene feature format*) and a FASTA
files to work on both genomes. A specific functional category will be
selected and we will compute:

-   The number of tRNAs of our sample.
-   The number of rRNAs of our sample.
-   The number of genes encoding proteins without known function.

Finally, using *RStudio* we will create a barplot of the types of genes
found in our MAG and our genome of reference (type strain).

## [Practical 7: Genome annotation II]{.underline}

### 1. Multiple sequence alignment

Both nucleotide and amino acid sequences from our MAG and the selected
reference genome will be aligned with MAFFT (Multiple Alignment using
Fast Fourier Transform). The resulting file containing the alignment of
both sequences will be visualized with Jalview.

### 2. Pairwise sequence alignment

In the same way as in the previous step, the pairwise alignment socre
for both nucleotide and amino acid sequences from our MAG and the
reference genome will be assessed using SMS (Sequence Manipulation
Suite) from *bioinformatics.org*. This parameter takes into account the
number of matches (i.e. A-A) and missmatches (i.e. A-T) between both
sequences (as well as the possible gaps present in one of them) and
computes them in order to give a value referring to their alignment
level.

In our case Point accepted mutation (PAM) will be used to score the
sequence alignments. PAM measures the amount of evolutionary distance
between two amino acid sequences by the construction of PAM matrices. It
uses PAM units matrices to determine such distance; for instance, one
PAM unit matrix (PAM1 matrix) represents substitution probabilities for
sequences that have experienced a single point mutation per 100 amino
acids.

It is important to mention that PAM requires proteins which have high
similarity with their predecessors. Moreover, PAM250 is frequently used
for sequence comparison.

## [Practical 8: Building a phylogenetic tree]{.underline}

This last practical included in the project promotes the construction of
a phylogenetic tree. A protein will be selected from the MAG's
functional categories. Using a text editor, we will add a header of the
gene code and protein name for its identification. Afterwards, its
analysis will begin.

### 1. Homology search

[Protein Blast](https://blast.ncbi.nlm.nih.gov/Blast.cgi) will be used
to search for protein sequence homologies with other characterized
peptides. Among those, only 20 will be selected (from different species
and genus) and downloaded. Such sequences will be stored in the same
text file storing our reference protein.

### 2. Alignment and trimming

As we did in *Practical 7* MAFFT will be used to align all our 21
sequences. Then *Jalview* will be used to open the alignment and colour
its residues. Observing the alignment some gappy regions might be
identified. If so, they must be eliminated by trimming using *Trimal*
software. The trimmed output will be also observed in *Jalview*.

### 3. Tree inference and visualisation

*IQ-TREE* software will be selected to construct a phylogenetic tree of
the sequence aligned and trimmed file. Then,
[iTol](https://itol.embl.de/) will be used to visualize and modify the
outputted tree.

All bootstrap values will be selected to be displayed and our reference
protein will be highlighted.

# RESULTS AND DISCUSSION

## [Practical 4]{.underline}

Upon acquiring the raw data for analysis, we conducted a thorough
quality control assessment of all base pair positions within the short
reads. Remarkably, we observed consistently high-quality scores across
the entire span of the reads, including both head and tail fragments as
depicted in *Figure 1*. Consequently, a decision was made to forgo
Trimming procedures, aiming to preserve information from the extremities
of the short reads.

```{r Figure 1, echo= FALSE, fig.align = 'center', out.width = "30%", out.extra='trim={0 0.4cm 0 0.4cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 1.} Quality check reads scores"}
knitr::include_graphics(here::here("Quality_check_reads_Pr4.png"))
```

Following genome assembly, seven distinct bins were generated (*Figure
2*). Notably, bins 01, 03, and 04 exhibited completeness levels nearing
maximum values, approximating 100%. This indicates the presence of all
anticipated genes within the Metagenome-Assembled Genomes (MAGs).

```{r Figure 2, echo= FALSE, fig.align = 'center', out.width = "60%", fig.cap = "\\scriptsize \\textbf{Figure 2.} Bins generated after metagenome assembly"}
knitr::include_graphics(here::here("output_bins_Pr4.png"))
```

Additionally, these bins demonstrated a contamination level below 1%,
affirming the appropriateness of the assembly process due to the initial
high-quality short reads, enabling seamless assembly (*Figure 3*). Thus,
bins 01, 03, and 04 can be considered of high quality, attaining \>50%
completeness and \<5% contamination.

Upon analyzing the distribution of the short reads utilized for MAG
assembly, it was observed that these ranged between 145-152 base pairs,
slightly longer than the optimal size for assembly. Furthermore,
approximately 5% of the sequences were identified as being repeated more
than 10 times. Considering these findings, the GC content of our MAG
might slightly deviate from the theoretical distribution (*Figure 4*).

```{r Figure 3 and 4, echo= FALSE, fig.align = 'center', out.width = "80%", out.extra='trim={0 2.5cm 0 2.5cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 3 and 4.} Bins completeness and contamination; GC content in bin 3"}
library(png)
library(grid)
library(gridExtra)
img1 <-  rasterGrob(as.raster(readPNG("completness_contamination_bins_Pr4.png")), interpolate = FALSE)
img2 <-  rasterGrob(as.raster(readPNG("GC_content_Pr4.png")), interpolate = FALSE)
grid.arrange(img1, img2, ncol = 2)


```

## [Practical 5]{.underline}

We opted to focus on bin03 due to its exceptional completeness level
(100%) and minimal contamination (0.91%). Following the annotation of
our MAG and subsequent genome set generation, we employed
[GTDB](https://gtdb.ecogenomic.org/) classification to determine the
bacterial strain origin of this genome. The outcomes are presented in
*Figure 5*, indicating a match between bin03 and *Bifidobacterium
dentium* with an Average Nucleotide Identity (ANI) score of 98.73, as
illustrated in the phylogenetic tree displayed in *Figure 6*. With a
gANI value higher than 95%, we can infer that our MAG originates from
*Bifidobacterium dentium*.

```{r Figure 5, echo= FALSE, fig.align = 'center', out.width = "100%", out.extra='trim={0 0.35cm 0 0.35cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 5.} Bacterial Taxonomy classification"}
knitr::include_graphics(here::here("especie_que_correspon_pr5.png"))
```

```{r Figure 6, echo= FALSE, fig.align = 'center', out.width = "60%", out.extra='trim={0 0.5cm 0 0.5cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 6.} Phylogenetic tree"}
knitr::include_graphics(here::here("arbre_detallat_rodonet_pr5.png"))
```

Subsequently, after obtaining the reference genome for *B. dentium*, we
conducted a comparative analysis with our MAG, computing the nucleotide
identity between both sequences. This assessment yielded an ANI distance
estimate of 98.72% (*Figure 7*), indicating a similarity of 98.72% in
nucleotide composition between the sequences. The comparison details can
be visualized in *Figure 8*.

```{r Figure 7, echo= FALSE, fig.align = 'center', out.width = "90%", out.extra='trim={0 0.2cm 0 0.2cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 7.} ANI distance estimation"}
knitr::include_graphics(here::here("ANI_estimate_ambos_pr5.png"))
```

```{r Figure 8, echo= FALSE, fig.align = 'center', out.width = "40%", out.extra='trim={0 0.2cm 0 0.2cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 8.} fastANI representation"}
knitr::include_graphics(here::here("rallas_rojas_raras_arribaRef.png"))
```

## [Practical 6]{.underline}

Upon comparing the feature counts between our genome (Bin03) and the
reference genome, we observed 2216 features in our genome as opposed to
2137 in the reference genome. Among these features, the functional
categories encoded in both genomes are depicted in *Figure* *9*.
Notably, both genomes exhibit similar genes in most categories, except
for: a) Stress response, b) Cell wall and capsule, c) RNA metabolism, d)
Membrane transport, and e) DNA metabolism.

```{r Figure 9, echo= FALSE, fig.align = 'center', out.width = "55%", out.extra='trim={0 0.28cm 0 0.28cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 9.} Functional categories encoded in MAG (left) and reference (right) genomes"}
knitr::include_graphics(here::here("tablas_genes_izq_bin3.png"))
```

Additionally, it is pertinent to ascertain the counts of tRNAs, rRNAs,
and hypothetical coding genes present in our MAG compared to the
reference. In our sample, the count stands at 46 for rRNAs, 6 for tRNAs,
and 1059 for putative coding genes. These counts are contrasted with
those of the reference in *Figure 10*.

```{r Figure 10,  echo=FALSE, fig.align = 'center', out.width = "30%", fig.show='hold',fig.cap = "\\scriptsize \\textbf{Figure 10.} MAG (left) and reference (right) genes graphic"}
library("ape")
library("ggplot2")

MAG_annotated <-read.delim("MAG.gff", header=F, comment.char="#") 
MAG_annotated.CDS <- MAG_annotated[MAG_annotated[,3]=="CDS",]

CDSinfo <- MAG_annotated.CDS[,9]

rRNA <- CDSinfo[grep("rRNA",CDSinfo)]
tRNA <- CDSinfo[grep("tRNA",CDSinfo)]                            
hypothetical <- CDSinfo[grep("hypothetical",CDSinfo)]

gene_type_table_mag <- data.frame(c("rRNA","tRNA","hypothetical","other_function"),
                                  c(length(rRNA),length(tRNA),length(hypothetical),length(CDSinfo)-length(rRNA)-length(tRNA)-length(hypothetical)))
colnames(gene_type_table_mag) = c("Class","Number")

REF_annotated <-read.delim("Reference.gff", header=F, comment.char="#") 
REF_annotated.CDS <- REF_annotated[REF_annotated[,3]=="CDS",]

CDSinfo <- REF_annotated.CDS[,9]

rRNA <- CDSinfo[grep("rRNA",CDSinfo)]
tRNA <- CDSinfo[grep("tRNA",CDSinfo)]                            
hypothetical <- CDSinfo[grep("hypothetical",CDSinfo)]

gene_type_table_ref <- data.frame(c("rRNA","tRNA","hypothetical","other_function"),
                                  c(length(rRNA),length(tRNA),length(hypothetical),length(CDSinfo)-length(rRNA)-length(tRNA)-length(hypothetical)))
colnames(gene_type_table_ref) = c("Class","Number")

ggplot(gene_type_table_mag, aes(x=Class, y=Number,fill=Class)) + 
  geom_bar(stat = "identity") 
ggplot(gene_type_table_ref, aes(x=Class, y=Number,fill=Class)) + 
  geom_bar(stat = "identity") 
```

## [Practical 7]{.underline}

The gene selected is categorized within the "Clustering-based
subsystems" gene group, known for its involvement in Bacterial Cell
Division. This gene group encompasses a total of 5 coding genes in both
our MAG and the reference genome. More precisely, the gene
IGLBDLCO_01642 found in the MAG and PKGELIAN_01503 identified in the
reference genome share a sequence length of 963 base pairs (bp). This
sequence encodes a protein comprising 320 aminoacids,known as the Cell
division protein FtsQ.

The multiple nucleotide sequence alignment between the MAG and the
reference genome is depicted in *Supplementary Info. 1*, revealing 9
mismatches across the 963 bp sequence length. These mismatches are
visually highlighted in *Figure 11*, displaying a gap in the consensus
sequence (black box). \vspace{20pt}

```{r Figure 11, echo= FALSE, fig.align = 'center', out.width = "100%", out.extra='trim={0 0.1cm 0 0.1cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 11.} Missmatches between FstQ reference (up) and MAG (down) nucleotide sequence."}
knitr::include_graphics(here::here("nt_GAP_alignmsequence.png"))
```

\vspace{70pt}

Notably, the multiple aminoacid alignment sequences is showcased in
*Supplementary info 2* and more explicitly in *Figure 12*. Among the 9
previous mismatches, it's evident that only 2 involve a non-synonymous
mutation in our MAG when compared to the reference:

-   I73V: Considering the non-polar nature of both aminoacids, the
    substitution of isoleucine with valine is unlikely to significantly
    impact the functionality of the protein.

-   P249S: Proline is non-polar, whereas serine is polar. This
    substitution may result in a change in functionality.

```{r Figure 12, echo= FALSE, fig.align = 'center', out.width = "100%", out.extra='trim={0 0.2cm 0 0.2cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 12.} Non-synonymous mutatuions between FstQ MAG (up) and reference (down) aminoacid sequence."}
knitr::include_graphics(here::here("aa_MISSMATCH_alignmseq.png"))
```

Finally, we conducted a pairwise nucleotide sequence alignment (not
depicted) using the following values: match = 2, mismatch = -1 and gap =
-2. This analysis yielded a score of 1899. Considering the sequence
length of 963 nucleotides (nt), the maximum attainable score would be
1926 if all nucleotides were identical. Additionally, we performed a
pairwise aminoacid sequence alignment using the PAM method (suitable for
closely related protein sequences), resulting in a score of 1993.

Based on these findings, it can be inferred that our MAG sequence
encoding the FtsQ protein closely resembles the reference sequence. This
suggests that our sample likely originates from *Bifidobacterium*
*dentium* or a microorganism sharing a similar FtsQ protein sequence.

## [Practical 8]{.underline}

Following the identification of 20 homologous sequences corresponding to
our FstQ sequence from *B. dentium*, a multiple sequence alignment was
conducted and subsequently refined by removing regions containing gaps
within our sequences. The resulting trimmed output is depicted in
*Figure 13*.

```{r Figure 13, echo= FALSE, fig.align = 'center', out.width = "100%", out.extra='trim={0 0.2cm 0 0.2cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 13.} Multiple FstQ sequence alignment after trimming gappy regions."}
knitr::include_graphics(here::here("protein_multialign.png"))
```

Ultimately, a phylogenetic tree encompassing these 21 sequences
(comprising 20 from BLAST results along with our sequence) has been
generated, as displayed in *Figure 14*. Notably, our FstQ sequence
demonstrates its closest relation to the reference FstQ sequence sourced
from *B. dentium* (WP 003838091.1), highlighting its affiliation with
this particular bacterium rather than with other type strains included
in the alignment.

```{r Figure 14, echo= FALSE, fig.align = 'center', out.width = "60%", out.extra='trim={0 0.2cm 0 0.2cm},clip', fig.cap = "\\scriptsize \\textbf{Figure 14.} Phylogenetic tree of the 21 aligned FtsQ sequences from different type strains."}
knitr::include_graphics(here::here("new_tree_PR8.png"))
```

\vspace{50pt}

# CONCLUSIONS

After conducting a comprehensive analysis across all five practical
experiments, a series of conclusions have been derived from our sample.

Initially, in practical 4, our sequence underwent computation and
scrutiny to assess its quality and potential integration into a cohesive
genome. High quality was observed in numerous reads, negating the
necessity for trimming and allowing for direct assembly. This led to the
creation of several contigs, resulting in the formation of 7 bins. Among
these bins, only the third met both completeness and contamination
criteria.

Subsequently, utilizing the selected bin and comparing it to a reference
genome facilitated the identification of the sample's origin. The whole
genome assembly was confirmed as belonging to *Bifidobacterium dentium*.
*Bifidobacterium*, a genus of gram-positive, non-motile, anaerobic
bacteria, predominantly resides in the gastrointestinal tract of
mammals, although certain strains have been isolated from both vaginal
and oral microbiota. Their colonization in the human intestine is
associated with advantageous health effects, including immune system
development, neuromodulation, and microbiotal equilibrium. Notably,
*Bifidobacterium dentium* inhabits both intestinal and oral environments
and has been linked to goblet cell maturation, serotonin, and GABA
production, thereby regulating the gut-brain-axis.

Subsequent examination of the genome revealed a diverse array of
physiological functions categorized across several identified genes.
Finally, to confirm the origin of our sample as *B.* *dentium*, a
comparative analysis of a protein sequence encoded by our genome was
performed against a reference sequence, validating the bacterial
strain's identity.

# REFERENCES

1.  GTDB - Genome Taxonomy Database [Internet]. [cited 2023 Nov].
    Available from: <https://gtdb.ecogenomic.org/>
2.  Letunic I, Bork P. Interactive Tree Of Life (iTOL) v5: an online
    tool for phylogenetic tree display and annotation. Nucleic Acids
    Research. 2021 Jul 2;49(W1):W293--6.
3.  Waterhouse AM, Procter JB, Martin DMA, Clamp M, Barton GJ. Jalview
    Version 2---a multiple sequence alignment editor and analysis
    workbench. Bioinformatics. 2009 May 1;25(9):1189--91.
4.  Katoh K, Rozewicki J, Yamada KD. MAFFT online service: multiple
    sequence alignment, interactive sequence choice and visualization.
    Briefings in Bioinformatics. 2019 Jul 19;20(4):1160--6.
5.  Point accepted mutation. In: Wikipedia [Internet]. 2023 [cited 2023
    Nov]. Available from:
    <https://en.wikipedia.org/w/index.php?title=Point_accepted_mutation&oldid=1184343745>
6.  KBase [Internet]. [cited 2023 Nov]. Welcome to KBase Predictive
    Biology. Available from: <https://www.kbase.us/>

\newpage

# SUPPLEMENTARY INFO

```{r Supp. 1, echo= FALSE, fig.align = 'center', out.width = "45%", out.extra='trim={0 0.1cm 0 0.1cm},clip', fig.cap = "\\scriptsize \\textbf{Supplementary Info. 1} FstQ nucleotide sequence alignment between MAG (up) and reference (down)."}
knitr::include_graphics(here::here("pairwase_nt_alignm.png"))
```

```{r Supp. 2, echo= FALSE, fig.align = 'center', out.width = "45%", out.extra='trim={0 0.1cm 0 0.1cm},clip', fig.cap = "\\scriptsize \\textbf{Supplementary Info. 2} FstQ amino acid sequence alignment between MAG (up) and reference (down)."}
knitr::include_graphics(here::here("pairwase_aa_alignm.png"))
```
